idelib/transforms.py

```diff
@@ -219,18 +270,29 @@ def __init__(self, coeffs, calId=None, dataset=None, reference=0,
             :keyword attributes: A dictionary of generic attributes, e.g.
                 ones parsed from `Attribute` EBML elements, or `None`.
         """
+        if len(coeffs) > 2 or len(coeffs) == 0:
+            raise ValueError('Coefficients of length {} are not supported.  '
+                             'Coefficients must be length 1 (constant) or 2 '
+                             '(linear).'.format(len(coeffs)))
+        coeffs = tuple(float(x) for x in coeffs)
         self.id = calId
         self.dataset = dataset
         self._coeffs = tuple(coeffs)
+        self._fastCoeffs = tuple(coeffs)
         self._variables = (varName,)
         self._references = (reference,)
         self._session = None
         self._lastSession = None
         self._timeOffset = 0
+        self._watchers = weakref.WeakSet()
         
         self.attributes = attributes
         
         self._build()
+
+
+    def __hash__(self):
+        return hash((self._coeffs, self._references))
```

Comment:
The coefficients and references won't be unique, even in the same file... should you at least check the ID as well?

Reply:
I've based the hash function(s) on the recommendation of the recommendation of the [Python docs for overriding the __hash__ function](https://docs.python.org/3/reference/datamodel.html#object.__hash__).  Basically, the recommendation is that the hash function combines the things that are compared in the `__eq__` method to generate a new hash, so that any objects that are equal to each other have the same hash.

Reply:
I thought the ID was part of the `__eq__()` comparison, but that doesn't appear to be the case. It really seems like should be.

Reply:
That would definitely make sense.  Do you want me to add both those changes?

Reply:
Yes, I think it is a good addition, although I don't know what effect it will have on the tests. I think this `__eq__()` was added when other classes got them for the sake of testing.